scipy/linalg/matfuncs.py

```diff
@@ -668,3 +675,329 @@ def rounded_sign(x):
         return S0
     else:
         return S0, errest
+
+
+def is_matrix_hermitian(A):
+    """
+    Returns True if a matrix is hermitian
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is hermitian.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_hermitian
+    >>> m = np.array([[0,1j],[-1j,0]])
+    >>> print(is_matrix_hermitian(m))
+    True
+    >>> m = np.array([[5,0],[1,3]])
+    >>> print(is_matrix_hermitian(m))
+    False
+    """
+    A = _asarray_square(A)
+    return np.sum(A == A.conj().T) == (A.shape[1]**2)
+
+
+def is_matrix_symmetric(A):
+    """
+    Returns True if a matrix is symmetric
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is symmetric.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_symmetric
+    >>> m = np.array([[2,3],[3,2]])
+    >>> print(is_matrix_symmetric(m))
+    True
+    >>> m = np.array([[5,0],[1,3]])
+    >>> print(is_matrix_symmetric(m))
+    False
+    """
+    A = _asarray_square(A)
+    return np.sum(A == A.T) == (A.shape[1]**2)
+
+
+def is_matrix_skew_symmetric(A, tol=1e-8):
+    """
+    Returns True if a matrix is skew symmetric
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    tol : float
+        Tolerance for the zero value.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is skew symmetric.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_skew_symmetric
+    >>> m = np.array([[5,1],[1,3]])
+    >>> print(is_matrix_skew_symmetric(m))
+    True
+    >>> m = np.array([[0, 2, -1],[-2, 0, -4],[-1, -4, 0]])
+    >>> print(is_matrix_skew_symmetric(m))
+    True
+    """
+    A = _asarray_square(A)
+    return (np.absolute(A - A.T) < tol).all()
+
+
+def is_matrix_nonsingular(A, tol=1e-8):
+    """
+    Returns True if a matrix is nonsingular
+    AKA invertible or nondegenerate
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    tol : float
+        Tolerance for the zero value.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is nonsingular.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_nonsingular
+    >>> m = np.array([[-1,1.5],[1,-1]])
+    >>> print(is_matrix_nonsingular(m))
+    True
+    >>> m = np.array([[1, -1, 0],[-1, 5, 0],[0, 0, 7]])
+    >>> print(is_matrix_nonsingular(m))
+    True
+    """
+    A = _asarray_square(A)
+    detA = det(A)
```

Comment:
Very very costly and not robust. A user can also do this very easily. The standard way to do this is estimating the condition number as it is, say, done with gecon, sycon, and pocon in the `scipy.linalg.solve()`.

Reply:
Fixed using cond as you suggested.